Sources/AblyChat/Messages.swift

```diff
@@ -54,21 +54,43 @@ public struct QueryOptionsWithoutDirection: Sendable {
 public struct MessageSubscription: Sendable, AsyncSequence {
     public typealias Element = Message
 
-    public init<T: AsyncSequence>(mockAsyncSequence _: T) where T.Element == Element {
-        fatalError("Not yet implemented")
+    private var subscription: Subscription<Element>
+
+    private typealias MockGetPreviousMessages = @Sendable (QueryOptionsWithoutDirection) async throws -> any PaginatedResult<Message>
+    private var mockGetPreviousMessages: MockGetPreviousMessages?
+
+    internal init(bufferingPolicy: BufferingPolicy) {
+        subscription = .init(bufferingPolicy: bufferingPolicy)
+    }
+
+    public init<T: AsyncSequence & Sendable>(mockAsyncSequence: T, mockGetPreviousMessages _: (QueryOptionsWithoutDirection) async throws -> any PaginatedResult<Message>) where T.Element == Element {
```

Comment:
I wonder what's the difference between `mockGetPreviousMessages _:` and `_ mockGetPreviousMessages:`

Reply:
It wasn't meant to be either of them 😅  — have fixed

And I had accidentally called my test `funcTestMockGetPreviousMessages` which meant it wasn't getting run and I didn't notice the above; have fixed that too